tests/PreparedTest.cpp

```diff
@@ -54,6 +54,7 @@ void testPreparedStatementResult (sql::connection& db)
         // explicit result scope
         // if results are released update should execute without exception
         auto result = db(preparedSelectAll);
+        assert(result.size() == 0);
```

Comment:
The next line "proves" that there is at least one result row. So size should be 1.

See: https://dev.mysql.com/doc/c-api/5.7/en/mysql-stmt-num-rows.html

Since ` mysql_stmt_store_result()` is never called, the size is not correct for bind_result, i.e. size() for bind_result needs to be removed.

And if we ever change the implementation of char_result in such a way that `mysql_store_result()` is not called, similar would be true for char_result.